src/main/scala/com/microsoft/hyperspace/index/rules/RuleUtils.scala

```diff
@@ -246,18 +268,33 @@ object RuleUtils {
           }
         }
 
+        val readSchema = if (filesDeleted.isEmpty) {
+          StructType(index.schema.filter(baseRelation.schema.contains(_)))
+        } else {
+          StructType(index.schema)
+        }
+
         val newLocation = new InMemoryFileIndex(spark, filesToRead, Map(), None)
         val relation = HadoopFsRelation(
           newLocation,
           new StructType(),
-          StructType(index.schema.filter(baseRelation.schema.contains(_))),
+          readSchema,
           if (useBucketSpec) Some(index.bucketSpec) else None,
           new ParquetFileFormat,
           Map())(spark)
 
         val updatedOutput =
           baseOutput.filter(attr => relation.schema.fieldNames.contains(attr.name))
-        baseRelation.copy(relation = relation, output = updatedOutput)
+
+        if (filesDeleted.isEmpty) {
+          baseRelation.copy(relation = relation, output = updatedOutput)
+        } else {
+          val lAttr = AttributeReference(IndexConstants.DATA_FILE_NAME_COLUMN, StringType)(
```

Comment:
`AttributeReference(IndexConstants.DATA_FILE_NAME_COLUMN, StringType)()` (`NamedExpression.newExprId` is default).

also what is `l` in `lAttr`?

Reply:
`l` from lineage. How about `lineageAttr` ??